trace/trace.go

```diff
@@ -275,6 +330,18 @@ func (s *Span) AddField(key string, val interface{}) {
 	}
 }
 
+func (s *Span) AddFieldIfUnset(key string, val any) {
```

Comment:
I'm unsure what the purpose of this function is? It looks like a direct copy of the `AddField` func above except it uses an `any` instead of `interface{}`. The function name implies it will only set the field if it doesn't already exist.